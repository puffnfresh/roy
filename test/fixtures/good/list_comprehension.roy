// Some simple comps.
console.log ([x | x ← [1, 2, 3]])
console.log ([x | x ← [1, 2, 3], x > 1])
console.log ([x + y | x ← [1, 2, 3], let y = 3, x > 1, y != x])
console.log ([x + y | x ← [2..5], y ← [1..3]])
console.log ([x + y + z | x ← [1, 2, 3], let yes = 'no', y ← [1, 2, 3], z ← [1, 2, 3]])
console.log ([word | word ← 'hello'])
console.log ([char | char ← ['h','e','l','l','o']])

// More complex ones.
// This should look up the scope and find `y` for the first generator.
// The first `w` should be shadowed by the second.
// Currently, neither of those happen
// let y = [1, 2]
// let z = 42
// console.log ([(x, y, z, w) | x ← y, w ← ['a', 'b'], y ← [4, 5], w ← ['c', 'd']])
console.log ([3 | true])
console.log ([f 37 | f ← [λx → x + 5, λid → id]])
