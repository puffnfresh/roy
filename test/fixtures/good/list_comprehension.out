[ 1, 2, 3 ]
[ 2, 3 ]
[ 5 ]
[ 3, 4, 5, 4, 5, 6, 5, 6, 7, 6, 7, 8 ]
[ 3, 4, 5, 4, 5, 6, 5, 6, 7, 4, 5, 6, 5, 6, 7, 6, 7, 8, 5, 6, 7, 6, 7, 8, 7, 8, 9 ]
[ 'hello' ]
[ 'h', 'e', 'l', 'l', 'o' ]
[ 3 ]
[ 42, 37 ]
